templates/web/base/report/form/user_loggedout_by_email.html

```diff
@@ -25,19 +23,44 @@
     </div>
   [% END %]
 
-    [% PROCESS 'report/form/user_loggedout_email.html' name='username_register' %]
+  [% IF type != 'update' AND c.config.SMS_AUTHENTICATION AND NOT c.cobrand.call_hook('disable_phone_number_entry') %]
 
-  [% IF type != 'update' AND c.config.SMS_AUTHENTICATION %]
-    [% UNLESS c.cobrand.call_hook('disable_phone_number_entry') %]
-        <div id="js-hide-if-username-phone" class="hidden-js">
-            <label for="form_phone">[% loc('Phone number (optional)') %]</label>
-            <input class="form-control" type="text" value="[% phone | html %]" name="phone" id="form_phone">
-        </div>
+  <fieldset>
+      <legend>How would you like to receive updates?</legend>
+
+    [% IF field_errors.update_method %]
+        <p class='form-error'>[% field_errors.update_method %]</p>
     [% END %]
-    <div id="js-hide-if-username-email" class="hidden-js">
-        <label for="form_email">[% loc('Email address (optional)') %]</label>
-        <input class="form-control" type="text" value="[% email | html %]" name="email" id="form_email">
-    </div>
+    <p class="segmented-control segmented-control--radio">
+        <input type="radio" name="update_method" id="update_method_email"
+             data-show="#js-optional-phone" data-hide="#js-optional-email"
+            value="email"[% ' checked' IF update_method == 'email' %]>
+        <label class="btn" for="update_method_email">[% loc('Email') %]</label>
+        <input type="radio" name="update_method" id="update_method_phone"
+             data-show="#js-optional-email" data-hide="#js-optional-phone"
+            value="phone"[% ' checked' IF update_method == 'phone' %]>
+        <label class="btn" for="update_method_phone">[% loc('Phone') %]</label>
+    </p>
```

Comment:
This doesn't change the form validation so you can select phone here, not fill in a phone number but fill in an email and it submits fine. Similarly if you don't fill in anything it says you need an email even though the label says optional.

Reply:
I can't reproduce this - if I select phone, leave phone blank, fill in email, the page reloads and the error is "Please enter your email" as per your second point.

Reply:
Oh you mean submits fine as in it doesn't change the HTML5 required attributes, right.

Reply:
Yes, I could have been a bit clearer there :)

Reply:
Have added JS bit that should set required as needed.

Reply:
I'm not sure what's happening with this but that doesn't seem to be working. The code is definitely running but doesn't seem to be having an effect. Also, if the JS validation doesn't kick in then the non JS validation is getting quite confused:

<img width="443" alt="Screenshot 2020-09-25 at 11 34 32" src="https://user-images.githubusercontent.com/5310/94257363-19340800-ff23-11ea-8d7d-ef6cb61941a4.png">

Ah, it looks like `fem.attr('required', true);` is the correct incantation.

I also think the display of the validation for the "How would you like to receive updates" bit needs some design love:

<img width="457" alt="Screenshot 2020-09-25 at 11 25 50" src="https://user-images.githubusercontent.com/5310/94257463-3f59a800-ff23-11ea-9c01-6e5e75e9dabb.png">

Reply:
Oh, this is me changing it to be jquery in the JS rather than direct in the HTML but not checking it after that, sorry.

The double error - yeah, I saw that happened but I didn't really want to change it, it seemed... hard. It's technically not confused because it dates from 01311af63412f39b75ff3b12bb6dc4051eab13b3 where a staff user using report-as-another to only enter a bad phone number in which case the error message needs to be attached to 'phone' field, but if someone is logging in with a bad phone number, needs to attach to 'username' field - and so here as it displays username error at top and then phone error by phone you get it twice. I couldn't see an easy way to fix this.

Will fix the required thing, whoops.